### ISSUE TITLE
Namespace Drop-Down Default to Null on New Variable Creation UI

### ISSUE THREAD

**Issue by ChefAustin** [hashicorp/nomad#24314]
### Proposal
When creating a new variable via UI (`ui/variables/new[?path=nomad%2Fjobs%2F]`), I would like to propose two small changes:
1. the pre-selected value for the namespace drop-down menu is an empty/blank/null namespace
2. the 'Save Variables' button enforces validation of the selection from the namespace drop-down menu before saving and then does not proceed until a namespace value is explicitly chosen (thereby forcing the user to choose a namespace)

### Use-cases
Often times I'll go to create a new variable and, in doing so, I'll forget to explicitly choose a namespace for it (thereby defaulting to whatever existing namespace is alphabetically first). Then I'll create populate the new variable with its necessary kv-pairs. Then I'll perform this same process again in our federated region. Finally I'll repeat this entire process again in our dev environment. 

This is when it hits me: "Blast! _::shakes fist at sky::_ I just created this var in the wrong namespace." Then I trace back my steps, delete the var (in all envs and regions) and recreate it in the proper namespace.

### Alternate Proposal
Allow for easily moving a Nomad Variable from one namespace to another.

### Attempted Solutions
Only workaround for this is for me to not be such an idiot (hard sell; trust me: I've tried 😆).

**Comment by gulducat** [comment]
Heya, thanks for the report!  I can see how that could be annoying (and I really wish I didn't relate so much to your Attempted Solutions :yum:).

Personally, for a repetitive task like this, I might instead write a little script with [`nomad` CLI](https://developer.hashicorp.com/nomad/docs/commands/var/put) or [HTTP API](https://developer.hashicorp.com/nomad/api-docs/variables/variables#create-variable) directly, or perhaps some Terraform with the [Nomad provider](https://registry.terraform.io/providers/hashicorp/nomad/latest/docs/resources/variable).

But I'll pass this over to UI folks to see what they think of the potential impact on other users who may actually prefer the current behavior (I tend to agree with you, but who knows what we might not be thinking of!)

**Comment by ChefAustin** [comment]
> Personally, for a repetitive task like this, I might instead write a little script with [`nomad` CLI](https://developer.hashicorp.com/nomad/docs/commands/var/put) or [HTTP API](https://developer.hashicorp.com/nomad/api-docs/variables/variables#create-variable) directly, or perhaps some Terraform with the [Nomad provider](https://registry.terraform.io/providers/hashicorp/nomad/latest/docs/resources/variable).

Agreed that a little shell sorcery would likely solve-for my issue here. Conversely, we do have teams here who might prefer using the UI over CLI for this task so I think the feature request is in-bounds but...
 
> But I'll pass this over to UI folks to see what they think of the potential impact on other users who may actually prefer the current behavior (I tend to agree with you, but who knows what we might not be thinking of!)

...I can also understand that any such implementation of this could be considered a user-muscle-memory-impacting-change (so I won't have my feelz hurt if it was deemed undesirable by the larger Nomad user community) and subsequently swatted-down.

All said, thanks for your consideration. 🫡
